sapl/legacy/migration.py

```diff
@@ -181,6 +182,225 @@ def garante_tabela_no_legado(create_table):
         assert existe_tabela_no_legado(tabela)
 
 
+def migra_autor():
+
+    SQL_ENUMERA_REPETIDOS = '''
+          select cod_parlamentar, COUNT(*)
+          from autor where cod_parlamentar is not null
+          group by cod_parlamentar
+          having 1 < COUNT(*)
+          order by cod_parlamentar asc;
+    '''
+
+    SQL_INFOS_AUTOR = '''
+          select cod_autor from autor
+          where cod_parlamentar = {}
+          group by cod_autor
+          order by col_username, des_cargo desc;
+    '''
+
+    SQL_UPDATE_AUTOR = "update autoria set cod_autor = {} where cod_autor in ({});"
+
+    SQL_ENUMERA_AUTORIA_REPETIDOS = '''
+        select cod_materia, COUNT(*) from autoria where cod_autor in ({})
+        group by cod_materia
+        having 1 < COUNT(*);
+    '''
+
+    SQL_DELETE_AUTORIA = '''
+        delete from autoria where cod_materia in ({}) and cod_autor in ({});
+    '''
+
+    SQL_UPDATE_DOCUMENTO_ADMINISTRATIVO = '''
+        update documento_administrativo
+        set cod_autor = {}
+        where cod_autor in ({});
+    '''
+
+    SQL_UPDATE_PROPOSICAO = '''
+        update proposicao
+        set cod_autor = {}
+        where cod_autor in ({});
+    '''
+
+    SQL_UPDATE_PROTOCOLO = '''
+        update protocolo
+        set cod_autor = {}
+        where cod_autor in ({});
+    '''
+
+    SQL_DELETE_AUTOR = '''
+        delete from autor where cod_autor in ({}) 
+        and cod_autor not in ({});
+    '''
+
+    cursor = exec_legado('update autor set ind_excluido = 0 where cod_autor is not null;')
+    cursor = exec_legado(SQL_ENUMERA_REPETIDOS)
+
+    autores_parlamentares = [r[0] for r in cursor if r[0]]
+
+    for cod_autor in autores_parlamentares:
+
+        sql = SQL_INFOS_AUTOR.format(cod_autor)
+
+        cursor = exec_legado(sql)
+        autores = []
+
+        for response in cursor:
+            autores.append(response)
```

Comment:
@sconetto 
Quase sempre que o padrão
```
a = []
for x in ...:
    a.append(x)
```

seria melhor escrito como: `a = [x for x in ...]`